airbyte-integrations/connectors/source-instagram/source_instagram/client/api.py

```diff
@@ -304,16 +306,23 @@ class MediaInsightsAPI(MediaAPI):
 
     def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         for account in self._api.accounts:
-            media = self._get_media(account["instagram_business_account"], {"limit": self.result_return_limit}, ["media_type"])
+            ig_account = account["instagram_business_account"]
+            media = self._get_media(ig_account, {"limit": self.result_return_limit}, ["media_type"])
             for ig_media in media:
-                yield {
-                    **{
-                        "id": ig_media.get("id"),
-                        "page_id": account["page_id"],
-                        "business_account_id": account["instagram_business_account"].get("id"),
-                    },
-                    **{record.get("name"): record.get("values")[0]["value"] for record in self._get_insights(ig_media)},
-                }
+                try:
+                    yield {
+                        **{
+                            "id": ig_media.get("id"),
+                            "page_id": account["page_id"],
+                            "business_account_id": ig_account.get("id"),
+                        },
+                        **{record.get("name"): record.get("values")[0]["value"] for record in self._get_insights(ig_media)},
+                    }
+                except FacebookRequestError as error:
```

Comment:
We should explicitly check the error code, we don't want to ignore all errors

Reply:
Agree, done